crates/matrix-sdk/src/attachment.rs

```diff
@@ -0,0 +1,285 @@
+use std::io::Read;
+#[cfg(feature = "image_proc")]
+use std::io::{BufRead, Seek};
+
+#[cfg(feature = "image_proc")]
+use image::GenericImageView;
+use ruma::{
+    assign,
+    events::room::{
+        message::{AudioInfo, FileInfo, VideoInfo},
+        ImageInfo, ThumbnailInfo,
+    },
+    UInt,
+};
+
+#[cfg(feature = "image_proc")]
+use crate::ImageError;
+
+/// Base metadata about an image.
```

Comment:
These are subsets of the info we get in each media event, without the data. This is what the user can provide with the attachment to build a complete event. I'm not sure if this would make sense in ruma, but we might have the types upstream if something like https://github.com/ruma/ruma/issues/790 is implemented